airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -48,6 +51,108 @@
 log = logging.getLogger(__name__)
 
 
+@router.patch(
+    "/{task_instance_id}/run",
+    status_code=status.HTTP_200_OK,
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Task Instance not found"},
+        status.HTTP_409_CONFLICT: {"description": "The TI is already in the requested state"},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {"description": "Invalid payload for the state transition"},
+    },
+)
+def ti_run(
+    task_instance_id: UUID, ti_run_payload: Annotated[TIEnterRunningPayload, Body()], session: SessionDep
+) -> TIRunContext:
+    """
+    Run a TaskInstance.
+
+    This endpoint is used to start a TaskInstance that is in the QUEUED state.
+    """
+    # We only use UUID above for validation purposes
+    ti_id_str = str(task_instance_id)
+
+    old = select(TI.state, TI.dag_id, TI.run_id).where(TI.id == ti_id_str).with_for_update()
+    try:
+        (previous_state, dag_id, run_id) = session.execute(old).one()
+    except NoResultFound:
+        log.error("Task Instance %s not found", ti_id_str)
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail={
+                "reason": "not_found",
+                "message": "Task Instance not found",
+            },
+        )
+
+    # We exclude_unset to avoid updating fields that are not set in the payload
+    data = ti_run_payload.model_dump(exclude_unset=True)
+
+    query = update(TI).where(TI.id == ti_id_str).values(data)
+
+    if previous_state != State.QUEUED:
```

Comment:
Will we need to change this for reschedule/resume-after-defer?

Reply:
added a TODO

Reply:
@amoghrajesh The question is does the scheduler change the state to queued for those two states before it can get to a worker (My gut says yes, the scheduler does this)

Reply:
Umm interesting, I need to check it out

Reply:
@kaxil worth checking this

Reply:
Yeah, it is the following

![image](https://github.com/user-attachments/assets/4780749a-fd73-4471-bc89-68aeebce5300)

Reply:
https://github.com/apache/airflow/blob/89fdc0326a0cf71d8e95155047092102695d00b2/airflow/models/dagrun.py#L1064

https://github.com/apache/airflow/blob/89fdc0326a0cf71d8e95155047092102695d00b2/airflow/ti_deps/dependencies_states.py#L27-L31